base/test.jl

```diff
@@ -492,11 +494,25 @@ function finish(ts::DefaultTestSet)
     end
     println()
     # Recursively print a summary at every level
-    print_counts(ts, 0, align, pass_width, fail_width, error_width, broken_width, total_width)
+    print_counts(ts, depth_pad, align, pass_width, fail_width, error_width, broken_width, total_width)
+end
+
+# Called at the end of a @testset, behaviour depends on whether
+# this is a child of another testset, or the "root" testset
+function finish(ts::DefaultTestSet)
+    # If we are a nested test set, do not print a full summary
+    # now - let the parent test set do the printing
+    if get_testset_depth() != 0
+        # Attach this test set to the parent test set
+        parent_ts = get_testset()
+        record(parent_ts, ts)
+        return
+    end
     # Finally throw an error as we are the outermost test set
-    if total != total_pass
+    #print_test_results(ts)
```

Comment:
You can dump it explicitly with `Base.Test.print_test_results(Base.Test.get_testset())` but yeah, it would be silent unless things fail. There's a reason I put WIP on this. I wish I could put bold in titles: WI**P**. Ideally I think we'd have a `verbose/quiet` flag you could pass depending if you want output or not. It would be good to just let packages use `test/runtests.jl` themselves, rather than hacking their own weird fake parallel testing together like happens now.